### PR TITLE
Refactor Config and Secrets 

### PR DESCRIPTION
**Pull request by gortegaMS**
# Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] My code follows the code style of this project.
- [x] I ran lint checks locally prior to submission.
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR

Refactor Config and Secrets 

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes [#372](https://github.com/retaildevcrews/wcnp/issues/372),[ #371](https://github.com/retaildevcrews/wcnp/issues/371)

### CODE REVIEW

File: src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs [hunk 1]
```diff
@@ -129,16 +134,73 @@ public async Task Invoke(HttpContext context)
...
+                return "OK";
+            }
+        }
+
+        // get the client IP address from the request / headers
+        private static string GetClientIp(HttpContext context, out string xff)
+        {
+            const string XffHeader = "X-Forwarded-For";
+            const string IpHeader = "X-Client-IP";
+
```
Comment (jofultz) [inline src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs:163]: Should we move these to a static class for constants that we can use globally?  Header strings seem like something that should be defined for broad usage.

File: src/LodeRunner.API/src/Services/LRAPICacheService.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+using LodeRunner.API.Data;
+using LodeRunner.API.Interfaces;
+using LodeRunner.API.Models;
+using LodeRunner.Services;
+using Microsoft.Azure.Documents;
+
+namespace LodeRunner.API.Services
+{
+    public class LRAPICacheService : ILRAPICacheService
+    {
```
Comment (jofultz) [inline src/LodeRunner.API/src/Services/LRAPICacheService.cs:17]: Is this intended to be the final form of the cache and cache service or is there another iteration?  When we chatted about it I was thinking we'd have generic methods that would internally create, put, and get from caches based on type.  So, LRAPICacheService.GetClientByClientStatusId(id) would end up calling something like cache.GetItem<ClientStatus>(id).

Let's chat about it.

**Review by jofultz — CHANGES_REQUESTED** [review]
There's a couple of notes.  Let's grab 30 minutes and discuss.

**Review by jofultz — APPROVED** [review]
Talked with Gerardo.  This is the first part for secrets and config only.  So, approving and will handle caching in the next iteration.

File: src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs (hunk 1 above)
Comment (gortegaMS) [reply src/LodeRunner.API/src/Handlers/RequestLogger/RequestLogger.cs:163]: I will do it as part of Caching Refactor task

File: src/LodeRunner.API/src/Services/LRAPICacheService.cs (hunk 2 above)
Comment (gortegaMS) [reply src/LodeRunner.API/src/Services/LRAPICacheService.cs:17]: this does not include Caching Layer refactoring, there is an other branch out there that I created off "great merge" to minimize merge conflicts. On "great merge" branch I just reorganized Caching , it's intend is only to refractor Config an secrets

**Pull request merged**